pkg/dynamicrp/testhost/host.go

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright 2023 The Radius Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testhost
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/google/uuid"
+	v1 "github.com/radius-project/radius/pkg/armrpc/api/v1"
+	"github.com/radius-project/radius/pkg/armrpc/hostoptions"
+	"github.com/radius-project/radius/pkg/dynamicrp"
+	"github.com/radius-project/radius/pkg/dynamicrp/server"
+	"github.com/radius-project/radius/pkg/sdk"
+	"github.com/radius-project/radius/pkg/ucp"
+	"github.com/radius-project/radius/pkg/ucp/config"
+	"github.com/radius-project/radius/pkg/ucp/dataprovider"
+	queueprovider "github.com/radius-project/radius/pkg/ucp/queue/provider"
+	secretprovider "github.com/radius-project/radius/pkg/ucp/secret/provider"
+	ucptesthost "github.com/radius-project/radius/pkg/ucp/testhost"
+	"github.com/radius-project/radius/test/integrationtest/testhost"
+	"github.com/stretchr/testify/require"
+)
+
+// TestHostOptions supports configuring the dynamic-rp test host.
+type TestHostOption interface {
```

Comment:
```suggestion
// TestHostOptions supports configuring the dynamic-rp test host.
type TestHostOptions interface {
```

Reply:
This is good as it is. See other uses in our code for:

- `Options` as a [parameter-object](https://refactoring.guru/introduce-parameter-object)
- `Option` as a [mechanism for optional-arguments](https://dev.to/kittipat1413/understanding-the-options-pattern-in-go-390c) to a function. 

A good way to rationalize this is that an `TestHostOption` gets to modify the `dynamicrp.Options`